app/controllers/Api/v1/posts_controller.rb

```diff
@@ -1,28 +1,23 @@
 class Api::V1::PostsController < ApplicationController
-  before_action :generate_token, only: :create
-  before_action :generate_salt, only: [:create]
-  before_action :destroy_exceed_posts, only: [:secret]
+  before_action :find_post, only: [:reveal, :secret]
 
   def create
     render json: {error: {error_type: "validation", error_message: "params missing"}}, status: 422 and return if post_params[:body].empty?
-    @post = Post.new(post_params)
-    @post.body = Ciphering.new(post_params[:body], post_params[:salty_password] || "").encrypt
-    @post.salty_password = @salt
-    @post.url_token =  @token
-    if @post.save
-      render json: {data: {url: @post.attributes}}, status: 200
+    @post = Post.create_new(post_params)
+    if @post[:status] == 200
+      render json: {data: @post[:post]}, status: @post[:status]
     else
-      render json: {error: {error_type: "validation", error_message: "params missing"}}, status: 422
+      render json: {error: {error_type: "validation", error_message: "params missing"}}, status: @post[:status]
     end
   end
 
   def reveal
-    @post = Post.where(id: params[:id],url_token: params[:token]).last
     if @post
-      result = SaltChecker.new(@post.salty_password,params[:salty_password] ).call
+      result = SaltChecker.new(@post["salty_password"],params[:salty_password] ).call
```

Comment:
I think you can create a global `ApplicationService` class and inherit from it -

```
# app/services/application_service.rb
class ApplicationService
  def self.call(*args, &block)
    new(*args, &block).call
  end
end

class SaltChecker < ApplicationService
end
```

Then you can use it like this - `SaltChecker.call()` This looks much cleaner.